lib/chef/resource/windows_adjoin.rb

```diff
@@ -0,0 +1,90 @@
+#
+# Author:: John Snow (<jsnow@chef.io>)
+# Copyright:: 2016-2018, John Snow
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "chef/resource"
+require "chef/mixin/powershell_out"
+
+class Chef
+  class Resource
+    class WindowsAdJoin < Chef::Resource
+      resource_name :windows_adjoin
+      provides :windows_adjoin
+
+      include Chef::Mixin::PowershellOut
+
+      description "Use the windows_adjoin resource to join a Windows Active Directory domain."
+      introduced "14.0"
+
+      property :domain_name, String,
+               description: "The FQDN of the AD domain to join.",
+               validation_message: "The 'domain_name' property must be a FQDN.",
+               regex: /.\../, # anything.anything
+               name_property: true
+
+      property :domain_user, String,
+               description: "The domain user to use to join the host to the domain.",
+               required: true
+
+      property :domain_password, String,
+               description: "The password for the domain user.",
+               required: true
+
+      property :ou_path, String,
+               description: "The path to the OU where you would like to place the host."
+
+      property :reboot, Symbol,
+               equal_to: [:immediate, :delayed, :never],
+               validation_message: "The reboot property accepts :immediate (reboot as soon as the resource completes), :delayed (reboot once the Chef run completes), and :never (Don't reboot)",
+               description: "Controls the system reboot behavior post domain joining. Reboot immediately, after the Chef run completes, or never. Note that a reboot is necessary for changes to take effect.",
+               default: :immediate
```

Comment:
still think this would be easier to read:

``
      property :reboot, Symbol, equal_to: [:immediate, :delayed, :never], default: :immediate,
               validation_message: "The reboot property accepts :immediate (reboot as soon as the resource completes), :delayed (reboot once the Chef run completes), and :never (Don't reboot)",
               description: "Controls the system reboot behavior post domain joining. Reboot immediately, after the Chef run completes, or never. Note that a reboot is necessary for changes to take effect.",
```

the long ones are the ones you need new lines for, and if the shorter ones don't exceed the lengths of the longest description you might as well put those all together on the first line.

Reply:
I might go shuffle them around post 14

Reply:
Also I think I'm going to single line all the descriptions and just assume we're all soft wrapping because we're not doing our work on an Apple IIe